### ISSUE TITLE
Crash on Pig (itemInHand)

### ISSUE THREAD

**Issue by inxomnyaa** [alejandroliu/bad-plugins#17]
Fatal error: Call to a member function getItemInHand() on null in C:\Users\Administrator\Desktop\Server\MinecraftPE\PocketMine-MP\plugins\Mobsters\src\aliuly\mobsters\i                                                                     diots\Pig.php on line 112

Call Stack:
    0.0201     408808   1. {main}() C:\Users\Administrator\Desktop\Server\MinecraftPE\PocketMine-MP\PocketMine-MP.phar:0
    0.0223     403904   2. require('phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmine/PocketMine.php') C:\Users\A                                                                     dministrator\Desktop\Server\MinecraftPE\PocketMine-MP\PocketMine-MP.phar:1
    0.1506     543976   3. pocketmine\Server->__construct() phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmine/Poc                                                                     ketMine.php:464
   15.7152   20741432   4. pocketmine\Server->start() phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmine/Server__3                                                                     2bit.php:1778
   15.7321   20757520   5. pocketmine\Server->tickProcessor() phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmine/S                                                                     erver__32bit.php:2178
 7976.2291   95971552   6. pocketmine\Server->tick() phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmine/Server__32                                                                     bit.php:2300
 7976.2438   95980624   7. pocketmine\Server->checkTickUpdates() phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmin                                                                     e/Server__32bit.php:2497
 7976.2529   95980656   8. pocketmine\level\Level->doTick() phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmine/Ser                                                                     ver__32bit.php:2339
 7976.2568   95980888   9. aliuly\mobsters\idiots\Pig->onUpdate() phar://C:/Users/Administrator/Desktop/Server/MinecraftPE/PocketMine-MP/PocketMine-MP.phar/src/pocketmi                                                                     ne/level/Level__32bit.php:695
 7976.2574   95982336  10. aliuly\mobsters\idiots\Pig->findTarget() C:\Users\Administrator\Desktop\Server\MinecraftPE\PocketMine-MP\plugins\Mobsters\src\aliuly\mobsters                                                                     \idiots\Pig.php:156

**Comment by astersx** [comment]
Only on pigs?.....

**Comment by inxomnyaa** [comment]
How would you hold a Pig in your hand?
There aren't even pig spawn eggs in the inventory

**Comment by astersx** [comment]
There is spawn eggs in inventory,  scroll down ,!!!

**Comment by astersx** [comment]
You can disamble  gettitemonhand() only if is null,  if(getitemonhand() == null){} else{ code that is making pig to run to carrot}

I will do this too :D

**Comment by inxomnyaa** [comment]
Again. There is no AI and there are just 3 spawn eggs: Squids, Zombie and Villager.
And they do not move.
I bet you use a plugin.

**Comment by inxomnyaa** [comment]
OH SORRY I THOUGHT THIS WAS ON IMAGICALMINE!
Yea.. But its a bug

**Comment by astersx** [comment]
Yep, i fixed pigs error,  but  plugin makes the game crash ...sometimg about level name,  wtf :(....
It cant be fixed becouse mobs make server crash.

**Comment by inxomnyaa** [comment]
Level name missing? Oh.. I know whats wrong. But i'll sleep now

**Comment by astersx** [comment]
Nope,  worlds are made with /mw  . Dang :(,  and just fixed pigs (+ +)  . I wanna pigs on my sv :(.
i put if hand is null then hand=0,  and worked like charm,  if hand = null pigs stop moving :3

**Comment by astersx** [comment]
This is the code that fix that error
if($pl->getInventory()->getItemInHand() == null){$hand=0;}else{
			$hand = $pl->getInventory()->getItemInHand();}
Becouse wen checking if you hold carrots is if hand != carrot,  so if is 0....  0 != carrot.
Plugin check only if u dont hold carrot,  so that pig can continue normal activity!,,
